rmgpy/data/kinetics/family.py

```diff
@@ -1602,6 +1604,24 @@ def apply_recipe(self, reactant_structures, forward=True, unique=True):
                 lowest_labels.append(min(labels))
             product_structures = [s for _, s in sorted(zip(lowest_labels, product_structures))]
 
+        # if applying the family in reverse and the template reactants restrict multiplicity, we need to make sure that
+        # a reverse product matches the multiplicity-constrained template reactant because the forward template products
+        # do not enforce the multiplicity restriction
+        if not forward and reactant_type == 'mol':
```

Comment:
If you do
```suggestion
        if not forward and isinstance(reactant_structure, Molecule):
```
could you do away with the reactant_type variable and remove the changes above?